### ISSUE TITLE
Native scrollbars decrease smooth scrolling performance

### ISSUE THREAD

**Issue by equeim** [QupZilla/qupzilla#2246]
QupZilla 2.1.1, Arch Linux. When native scrollbars are enabled, smooth scrolling is not so smooth, looks like like 30 fps.

**Comment by Vistaus** [comment]
Can't reproduce this on Solus. What are your hardware specs and are you running any other apps when you notice this? If so, which ones?

**Comment by equeim** [comment]
Video card is AMD Radeon R9 280X, OpenGL driver is radeonsi from Mesa 17.0.1, xf86-video-ati 7.8.99. The issue is present regardless of if I run any other apps.

**Comment by nowrep** [comment]
I can reproduce it, you can enable FPS counter in web inspector and it's clear that it is locked to 30fps. The problem is rendering the scrollbars (probably because QtWebEngine is QQuickWidget and that doesn't mix well with QtWidgets performance-wise), but I have currently no idea how to fix it.
I guess I'll need to disable the native scrollbars by default (or even completely remove this feature).

**Comment by cranes-bill** [comment]
>...(or even completely remove this feature).

I'll be so happy if you do it! :smiley:

**Comment by nowrep** [comment]
You can disable it if you don't like it.

**Comment by cranes-bill** [comment]
Sure, i know. That is the first think that i'm doing when starting QupZilla with new profile.

**Comment by equeim** [comment]
>> probably because QtWebEngine is QQuickWidget and that doesn't mix well with QtWidgets performance-wise

AFAIK, QWebEngineView is QWidget.

**Comment by nowrep** [comment]
> AFAIK, QWebEngineView is QWidget.

Not under the hood, it was QOpenGLWidget before and is now QQuickWidget to share the same code as  QtQuick version.

**Comment by innermous** [comment]
I double this.

**Comment by nowrep** [comment]
It's now disabled by default.

**Comment by kkofler** [comment]
IMHO, it would be a better idea to disable smooth scrolling by default and keep native scrollbars enabled by default if the 2 features conflict.

**Comment by iondissonance** [comment]
@kkofler +++

**Comment by nowrep** [comment]
No, it wouldn't be better. It doesn't affect only smooth scrolling, but the whole rendering. If the scrollbars are redrawn (because the scroll position changed), it halves the frame rate of the whole webview. So, even though it is most visible during smooth scrolling, it will affect also pages that implements smooth scrolling with javascript, ...

Unless I fix the main issue, it will stay as it is (native scrollbars disabled by default).

**Comment by iondissonance** [comment]
I have bad smooth scrolling, but ok native scrollbars.

**Comment by elfgoh** [comment]
Running 2.1.2 on Mac OS X, installed via homebrew. Laggy scrolling persists even after disabling animated scroll setting

**Comment by nowrep** [comment]
Disabling animated scrolling have no effect at all, it only makes it less noticeable. You need to disable native scrollbars.

**Comment by NickLion** [comment]
So it's not a new behavior, strange I haven't noticed this before. And nothing can be done for native scrolls at this moment… I do like native scrolls.

**Comment by SGOrava** [comment]
Hello,
I also have this problem when scrolling with touchpad and mouse pointer is pointed over the scrollbar because for some crazy reason smooth-scrolling is forced when scrolling with touchpad
